nixos/modules/services/hardware/intel-lpmd.nix

```
@@ -0,0 +1,527 @@
+      };
+
+      State = lib.mkOption {
+        default = [ ];
+        description = ''
+          List of "state" definitions.
+        '';
+
+        type = with lib.types; listOf (submodule perStateSubmodule);
+      };
+    };
+  };
+
+  xmlSubmodule = {
+    options = {
+      lp_mode_cpus = lib.mkOption {
+        default = null;
+        description = ''
+          The set of active CPUs when in LPM (low power mode).
+          Leave as `null` to let the daemon auto-detect this.
+
+          Can be specified as comma separating string, or as a range by using
+          '..' or '-'.
+        '';
+
+        example = "1,2,4..6,8-10";
+
+        type = with lib.types; nullOr str;
+      };
+
+      Mode = lib.mkOption {
+        default = "0";
+        description = ''
+          Specifies the way to migrate tasks to active CPUs
+          when LPM is active.
+
+          - '0': set *cpuset* to the active CPUs for cgroup v2 based systemd
+          - '1': isolate non-LPM CPUs so tasks are scheduled for LPM CPUs only
+          - '2': force idle injection to non-LPM CPUs and let scheduler handle LPM CPUs
+        '';
+
+        type =
+          with lib.types;
+          enum [
+            "0"
+            "1"
+            "2"
+          ];
+      };
+
+      PerformanceDef = lib.mkOption {
+        default = "-1";
+        description = ''
+          Specifies the default behaviour when the power setting
+          is set to **Performance**.
+
+          - '-1': never enter LPM
+          - '0': opportunistic LPM enter/exit based of HFI/Util request
+          - '1': always stay in LPM
+        '';
+
+        type =
+          with lib.types;
+          enum [
+            "-1"
+            "0"
+            "1"
+          ];
+      };
+
+      BalancedDef = lib.mkOption {
+        default = "-1";
+        description = ''
+          Specifies the default behaviour when the power setting
+          is set to **Balanced**.
+
+          - '-1': never enter LPM
+          - '0': opportunistic LPM enter/exit based of HFI/Util request
+          - '1': always stay in LPM
+        '';
+
+        type =
+          with lib.types;
+          enum [
+            "-1"
+            "0"
+            "1"
+          ];
+      };
+
+      PowersaverDef = lib.mkOption {
+        default = "-1";
+        description = ''
+          Specifies the default behaviour when the power setting
+          is set to **Power saver**.
+
+          - '-1': never enter LPM
+          - '0': opportunistic LPM enter/exit based of HFI/Util request
+          - '1': always stay in LPM
+        '';
+
+        type =
+          with lib.types;
+          enum [
+            "-1"
+            "0"
+            "1"
+          ];
+      };
+
+      HfiLpmEnable = {
+        default = false;
+        description = ''
+          Specifies if the HFI monitor can capture HFI hints for LPM.
+        '';
+
+        type = with lib.types; bool;
+      };
```

```suggestion
      HfiLpmEnable = {
        type = lib.types.bool;
        default = false;
        description = ''
          Specifies if the HFI monitor can capture HFI hints for LPM.
        '';
      };
```
we shouldn't use with lib here